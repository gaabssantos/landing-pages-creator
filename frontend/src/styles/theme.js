export const theme = {
  mainBg: 'red',
  secondaryBg: 'blue',
};
