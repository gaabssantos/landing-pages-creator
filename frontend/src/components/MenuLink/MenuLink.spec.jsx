import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { MenuLink } from '.';

describe('<MenuLink />', () => {
  it('should render a link', () => {
    renderTheme(<MenuLink link='https://youtube.com/'>Children</MenuLink>);
    const menu = screen.getByText('Children');

    expect(menu).toBeInTheDocument();
    expect(menu).toHaveAttribute('href', 'https://youtube.com/');
  });

  it('should render a link with target _self', () => {
    renderTheme(<MenuLink link='https://youtube.com/' newTab={false}>Children</MenuLink>);
    const menu = screen.getByText('Children');

    expect(menu).toBeInTheDocument();
    expect(menu).toHaveAttribute('target', '_self');
  });

  it('should render a link with target _blank', () => {
    renderTheme(<MenuLink link='https://youtube.com/' newTab={true}>Children</MenuLink>);
    const menu = screen.getByText('Children');

    expect(menu).toBeInTheDocument();
    expect(menu).toHaveAttribute('target', '_blank');
  });

  it('should match snapshot', () => {
    const {container} = renderTheme(<MenuLink link='https://youtube.com/' newTab={true}>Children</MenuLink>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
