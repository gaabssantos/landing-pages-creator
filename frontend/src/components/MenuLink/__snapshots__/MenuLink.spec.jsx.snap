// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  position: relative;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #DC143C;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<div>
  <a
    class="c0"
    href="https://youtube.com/"
    target="_blank"
  >
    Children
  </a>
</div>
`;
